src/main/java/de/tum/in/www1/artemis/service/CourseExportService.java

```diff
@@ -0,0 +1,304 @@
+            exportExercises(exam.getCourse().getId(), exercises, examDir.toString(), exportErrors);
+        }
+        catch (IOException e) {
+            var error = "Failed to create exam directory " + examDir + ".";
+            log.info(error);
+            exportErrors.add(error);
+        }
+    }
+
+    /**
+     * Exports the specified exercises by creating a zip file for each exercise.
+     * The zip files are created in the directory specified by outputDir. The function returns
+     * a list of exercise ids that couldn't be exported.
+     *
+     * @param exercises    The exercises to export
+     * @param outputDir    The path to a directory that will be used to store the zipped files.
+     * @param exportErrors List of failures that occurred during the export
+     */
+    private void exportExercises(long courseId, Set<Exercise> exercises, String outputDir, List<String> exportErrors) {
+        AtomicInteger exportedExercises = new AtomicInteger(0);
+        exercises.forEach(exercise -> {
+            // Notify the user after the progress
+            exportedExercises.addAndGet(1);
+            notifyUserAboutCourseExportState(courseId, CourseExportState.RUNNING, exportedExercises + "/" + exercises.size() + " done");
+
+            if (exercise instanceof ProgrammingExercise) {
+                programmingExerciseExportService.exportProgrammingExercise((ProgrammingExercise) exercise, outputDir, exportErrors);
+                return;
+            }
+
+            // Export the other exercises types
+
+            // Export options
+            var submissionsExportOptions = new SubmissionExportOptionsDTO();
+            submissionsExportOptions.setExportAllParticipants(true);
+
+            // The zip file containing student submissions for the other exercise types
+            Optional<File> exportedSubmissionsFileOrEmpty = Optional.empty();
+
+            try {
+                if (exercise instanceof FileUploadExercise) {
+                    exportedSubmissionsFileOrEmpty = fileUploadSubmissionExportService.exportStudentSubmissions(exercise.getId(), submissionsExportOptions);
+
+                }
+                else if (exercise instanceof TextExercise) {
+                    exportedSubmissionsFileOrEmpty = textSubmissionExportService.exportStudentSubmissions(exercise.getId(), submissionsExportOptions);
+
+                }
+                else if (exercise instanceof ModelingExercise) {
+                    exportedSubmissionsFileOrEmpty = modelingSubmissionExportService.exportStudentSubmissions(exercise.getId(), submissionsExportOptions);
+                }
+                else if (exercise instanceof QuizExercise) {
+                    // TODO: Quiz submssions aren't supported yet
+                    return;
+                }
+                else {
+                    // Exercise is not supported so return
+                    return;
+                }
```

Comment:
Remove the last else

```suggestion
            try {
                if (exercise instanceof FileUploadExercise) {
                    exportedSubmissionsFileOrEmpty = fileUploadSubmissionExportService.exportStudentSubmissions(exercise.getId(), submissionsExportOptions);
                }
                else if (exercise instanceof TextExercise) {
                    exportedSubmissionsFileOrEmpty = textSubmissionExportService.exportStudentSubmissions(exercise.getId(), submissionsExportOptions);
                }
                else if (exercise instanceof ModelingExercise) {
                    exportedSubmissionsFileOrEmpty = modelingSubmissionExportService.exportStudentSubmissions(exercise.getId(), submissionsExportOptions);
                }
                else if (exercise instanceof QuizExercise) {
                    // TODO: Quiz submssions aren't supported yet
                    return;
                }
                // Exercise is not supported so return
                return;
```

Reply:
This was an incorrect change because its an early return which would prevent any any exercise from being exported.